flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/utils/TestingResourceManagerGateway.java

```diff
@@ -309,6 +317,26 @@ public void disconnectJobManager(JobID jobId, Exception cause) {
 		}
 	}
 
+	@Override
+	public CompletableFuture<TransientBlobKey> requestTaskManagerFileUploadByName(ResourceID taskManagerId, String fileName, Time timeout) {
+		final Function<Tuple2<ResourceID, String>, CompletableFuture<TransientBlobKey>> function = requestTaskManagerFileUploadByNameFunction;
+
+		if (function != null) {
+			return function.apply(Tuple2.of(taskManagerId, fileName));
+		} else {
+			return CompletableFuture.completedFuture(new TransientBlobKey());
+		}
+	}
+
+	@Override
+	public CompletableFuture<Collection<Tuple2<String, Long>>> requestTaskManagerLogList(ResourceID taskManagerId, Time timeout) {
+		List<Tuple2<String, Long>> logsList = new ArrayList<>();
```

Comment:
This is not consistent with the other implementations. Moreover, we are not testing the handler's behavior if an exception is thrown by the gateway.